data/simpy.gram

```diff
@@ -363,11 +363,8 @@ atom[expr_ty]:
     | &'[' list
     | &'{' dict
     | &'[' listcomp
```

Comment:
Why not combine these three too?

Reply:
Because when we counted the number of each AST node in the stdlib we had found out that the amount of tuples, list and dicts was much greater than the amount of the atom nodes. If we were to combine these alternatives, wouldn't we lose time due to this?

Reply:
> Because when we counted the number of each AST node in the stdlib we had found out that the amount of tuples, list and dicts was much greater

Ideally, we should repeat those stats in the future with the PyPy packages to double-check those assumptions.

Reply:
I don't think so, because the lookahead ensures that we take exactly one path for each start character.

Reply:
> I don't think so, because the lookahead ensures that we take exactly one path for each start character.

Of course.